content/en/blog/_posts/2023-12-19-taint-eviction-controller.md

```diff
@@ -0,0 +1,80 @@
+---
+layout: blog
+title: "Kubernetes 1.29: Decouple taint-manager from node-lifecycle-controller"
+date: 2023-12-19
+slug: kubernetes-1-29-taint-eviction-controller
+---
+
+**Authors:** Yuan Chen (Apple), Andrea Tosatto (Apple)
+
+This blog discusses a new feature in Kubernetes 1.29 to improve the handling of taint-based pod eviction.
+
+## Background
+
+In Kubernetes 1.29, an improvement has been introduced to enhance the handling of taint-based pod eviction on nodes. This blog discusses the changes made to node-lifecycle-controller to separate its responsibilities and improve overall code maintainability.
+
+### Summary of changes
+
+node-lifecycle-controller previously combined two independent functions:
+
+- Adding a pre-defined set of `NoExecute` taints to nodes based on the node conditions.
+- Performing pod eviction on `NoExecute` taint.
+
+With the Kubernetes 1.29 release, the taint-based eviction implementation has been
+moved out of node-lifecycle-controller into a separate and independent component called taint-eviction-controller.
+This separation aims to disentangle code, enhance code maintainability,
+and facilitate future extensions to either component.
+
+As part of the change, additional metrics are introduced to monitor taint-based pod evictions.
+
+- `pod_deletion_duration_seconds` measures th latency between the time when a taint effect has been activated for the Pod and its deletion via taint-eviction-controller.
+- `pod_deletions_total` reports the total number of Pods deleted by taint-eviction-controller since its start.
+
+### How to use the new feature?
+
+A new feature gate, `SeparateTaintEvictionController`, has been added. The feature is enabled by default as Beta in Kubernetes 1.29.  
```

Comment:
We have already documented this gate, right?
Adding a link to the gate page would be desirable.

Reply:
Added a link to the feature gate reference. Note that a doc. update PR for the change is under review too. https://github.com/kubernetes/website/pull/43555/files.